bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -92,29 +128,18 @@ public void ensureChannelCount(int capacity) throws IOException {
     if (channels.get().size() < capacity) {
       synchronized (this) {
         if (channels.get().size() < capacity) {
-          List<ManagedChannel> newChannelList = new ArrayList<>(channels.get());
+          Counter counter = getActiveChannelCounter();
+          List<ChannelAndMetrics> newChannelList = new ArrayList<>(channels.get());
           while(newChannelList.size() < capacity) {
-            newChannelList.add(factory.create());
+            newChannelList.add(createChannelAndMetrics());
+            counter.inc();
```

Comment:
On second thought, I'm going to move the inc()/dec() responsibility to ChannelAndMetrics.